minifi/minifi-commons/minifi-commons-framework/src/main/java/org/apache/nifi/minifi/commons/service/FlowEnrichService.java

```diff
@@ -103,10 +107,12 @@ public byte[] enrichFlow(byte[] flowCandidate) {
             rootGroup.setInstanceIdentifier(randomUUID().toString());
         }
 
+        rootGroup.getControllerServices().forEach(cs -> cs.setScheduledState(ENABLED));
+
         Optional<VersionedControllerService> commonSslControllerService = createCommonSslControllerService();
         commonSslControllerService
             .ifPresent(sslControllerService -> {
-                List<VersionedControllerService> currentControllerServices = ofNullable(versionedDataflow.getControllerServices()).orElseGet(ArrayList::new);
+                List<VersionedControllerService> currentControllerServices = new ArrayList<>(versionedDataflow.getControllerServices());
```

Comment:
Since the initialization was moved outside now it is guaranteed that we will get a non-null list object here.
Is it necessary to recreate the list object for some reasons?
If not we could simplify the ifPresent block by reducing it to 
```
commonSslControllerService.ifPresent(versionedDataflow.getControllerServices()::add);
```

Reply:
I wanted to avoid passing the reference for the original list, but maybe you are right. As this object doesn't exist outside of this class we can safely use the value directly. I'm going to update this and the other as well.